File: app/src/main/java/com/util/base/MainPresenter.java
package com.util.base;

import android.content.Context;


public class MainPresenter extends MainContract.Presenter {

    /**
     * 构造方法
     *
     * @param context
     * @param view
     */
    public MainPresenter(Context context, MainContract.View view) {
        super(context, view);
    }

    @Override
    public void refreshToken() {

    }
}

File: app/src/main/java/com/util/base/ui/main/MainPresenter.kt
package com.util.base.ui.main

import android.content.Context
import com.data.repository.UserRepository

class MainPresenter(context: Context, view: MainContract.View) : MainContract.Presenter(context, view) {
    override fun login(phone: String, password: String) = launch {
        UserRepository.login(phone, password).collect {
            mView?.onLogin()
        }
    }

}